sympy/functions/elementary/hyperbolic.py

```diff
@@ -670,10 +670,10 @@ def _eval_as_leading_term(self, x):
         return (1/self._reciprocal_of(self.args[0]))._eval_as_leading_term(x)
 
     def _eval_is_real(self):
-        return self._reciprocal_of(args[0]).is_real
```

Comment:
What got uses for ```args``` before the change?
Just curious - this seems to be a trap for the unwary, and I'd like to see how widespread it is.